lib/inspec/dependencies/requirement.rb

```diff
@@ -94,17 +92,18 @@ def dependencies
     end
 
     def to_s
-      "#{name ? name : '<unfetched>'} (#{resolved_source})"
+      name
     end
 
     def profile
+      return @profile if ! @profile.nil?
+
       opts = @opts.dup
-      opts[:cache] = @cache
       opts[:backend] = @backend
       if !@dependencies.nil?
         opts[:dependencies] = Inspec::DependencySet.from_array(@dependencies, @cwd, @cache, @backend)
       end
-      @profile ||= Inspec::Profile.for_target(opts, opts)
+      @profile = Inspec::Profile.for_fetcher(fetcher, opts)
```

Comment:
I was thinking about this for a while, since it now deviated from our CLI code. We are using a separate mechanism for `inspec json` now.